### PR TITLE
feat(es/minifier): Merge assignments using sequential inliner

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

We can optimize `x = 1, x += 2` as `x = 3`.

**Related issue:**

### CODE REVIEW

File: crates/swc/tests/vercel/full/next-31419/1/output/index.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ Promise.all(assignAll).then(function() {
                         return n(this, function(n) {
                             switch(n.label){
                                 case 0:
-                                    return t = r[u], s += "'".concat(t.id, "', "), [
+                                    return s += "'".concat((t = r[u]).id, "', "), [
```
Comment (kdy1) [inline crates/swc/tests/vercel/full/next-31419/1/output/index.js:15]: This seems wrong

File: crates/swc_ecma_minifier/tests/fixture/next/wrap-contracts/output.js [hunk 2]
```diff
@@ -13616,14 +13616,14 @@
             }
             function sum64_4_hi(ah, al, bh, bl, ch, cl, dh, dl) {
                 var carry = 0, lo = al;
-                return carry += (lo = lo + bl >>> 0) < al ? 1 : 0, carry += (lo = lo + cl >>> 0) < cl ? 1 : 0, ah + bh + ch + dh + (carry += (lo = lo + dl >>> 0) < dl ? 1 : 0) >>> 0;
+                return carry = 0 + ((lo = lo + bl >>> 0) < al ? 1 : 0), carry += (lo = lo + cl >>> 0) < cl ? 1 : 0, ah + bh + ch + dh + (carry += (lo = lo + dl >>> 0) < dl ? 1 : 0) >>> 0;
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/tests/fixture/next/wrap-contracts/output.js:13619]: Minor regression?

File: crates/swc_ecma_minifier/src/compress/optimize/sequences.rs [hunk 3]
```diff
@@ -1315,20 +1315,23 @@ where
...
-                    a_id = tracing::field::debug(&a_id),
+                    a = tracing::field::debug(&a),
                     b = &*b_str
                 )
                 .entered(),
             )
         };
 
-        if b.is_class() || b.is_fn_expr() || b.is_arrow() {
+        if b.is_lit() || b.is_class() || b.is_fn_expr() || b.is_arrow() {
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/src/compress/optimize/sequences.rs:1334]: I'm confused by this change. `is_lit` seems like a shorthand for `is_literal`, and wouldn't we be able to optimize a literal value?

File: crates/swc_ecma_minifier/src/compress/optimize/sequences.rs [hunk 4]
```diff
@@ -1996,6 +2013,72 @@ where
...
+
+                        Box::new(a.take())
+                    }
+                }
+            };
+        }
+
+        // x = 1, x += 2 => x = 3
+        match b {
+            Expr::Assign(AssignExpr { op: op!("="), .. }) => {}
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/src/compress/optimize/sequences.rs:2059]: Could we drop `x = 1` in this case?

**Review by jridgewell — APPROVED** [review]

File: crates/swc_ecma_minifier/src/compress/optimize/sequences.rs (hunk 4 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/optimize/sequences.rs:2059]: Oh you are right, thanks!

**Review by jridgewell — APPROVED** [review]

File: crates/swc_ecma_minifier/tests/fixture/issues/5280/output.js [hunk 5]
```diff
@@ -1,4 +1,3 @@
 export function source() {
-    let c = 0, a = 1;
-    console.log(a += 5, c += 1, c);
+    console.log(a = 6, 1, 1);
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/fixture/issues/5280/output.js:2]: Oh this was wrong

**Pull request merged**